Custom Terms Test Changes - Create + List Experiment

bhanvimenghani opened the pull request:
## Description
This pr includes test changes for the Create Exp + List Exp api in **remote monitoring tests f**or custom terms functionality. 
https://docs.google.com/document/d/1yjavz5LnEWQOlGJZTED5YdEiCUNJJUqFr0uhhiJrsqE/edit?usp=sharing

Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Please describe the tests that were run to verify your changes and steps to reproduce. Please specify any test configuration required. 

- [ ] New Test X
- [ ] Functional testsuite

**Test Configuration**
* Kubernetes clusters tested on: 

## Checklist :dart:

- [ ] Followed coding guidelines
- [ ] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

## Additional information

Include any additional information such as links, test results, screenshots here

chandrams commented on tests/scripts/remote_monitoring_tests/rest_apis/test_create_experiment.py:
```diff
@@ -751,3 +751,558 @@ def test_create_exp_exptype_dates_columns(cluster_type):
+    # Note: JSON parsing will automatically deduplicate keys, keeping only the last occurrence
+    # This test validates the behavior - the experiment should be created with only one instance
+    # or fail depending on backend validation
+    assert response.status_code == SUCCESS_STATUS_CODE or response.status_code == ERROR_STATUS_CODE
```
Update the assertion as per the expected behavior

chandrams commented on tests/scripts/remote_monitoring_tests/rest_apis/test_list_experiment.py:
```diff
@@ -0,0 +1,138 @@
+def test_list_exp_mixed_terms_and_partial_auto_population(cluster_type):
+    """
+    Test Description: This comprehensive test validates auto-population behavior for:
+    1. Default terms (weekly, monthly) are auto-populated with system-defined values
```
Include a parameterized test to validate auto-population for all default terms

bhanvimenghani commented on tests/scripts/remote_monitoring_tests/rest_apis/test_create_experiment.py:
```diff
@@ -751,3 +751,558 @@ def test_create_exp_exptype_dates_columns(cluster_type):
+
+
+@pytest.mark.negative
+@pytest.mark.parametrize("plots_datapoint_value", [0, -1, -10])
```
Added the test in create experiment that checks if plots datapoint is positive but greater than the duration in days value, it is creating the experiment, need to make backend change to not allow experiment creation. For now i have added this test but adding skip option so that this does not runs untill i make backend changes

bhanvimenghani commented:
i have addressed the review comments and stashed the changes locally. Since there were changes in the json design the dev prs needs to be updated to support latest json structure and also incremental prs for it raised post that I will be changing the tests as required and updating this pr
